src/cpp/shared/game/game.cpp

```diff
@@ -2057,6 +2057,7 @@ int game::GetLuck(hero* hro, army *stack, town *castle) {
   auto res = ScriptCallbackResult<int>("OnCalcLuck", deepbind<hero*>(hro), deepbind<army*>(stack), deepbind<town*>(castle), luck);
   if(res.has_value())
     luck = res.value();
+  luck = max(-3, min(luck, 3));
```

Comment:
I believe luck is [-9, 9] or [-10,10], not [-3, 3].

Reply:
From original code
`    if ( totLuck < -3 )
      totLuck = -3;
    if ( totLuck > 3 )
      totLuck = 3;`

Reply:
Okay then